### DISCUSSION TITLE
Inverter polling frequency, but which file to edit ?

### DISCUSSION THREAD

**Discussion by TMoersch** [wlcrs/huawei_solar#161] [Setting up the integration]
Hi,
I want to changte the "Inverter polling frequency", but which file to edit?
I dont see any other option to correct the value via GUI, or did I miss something??

Thnaks To all
Thomas

**Comment by McFSR** [comment]
I did not try this myself, but look at const.py

UPDATE_INTERVAL = timedelta(seconds=30)

**Comment by RabAusten** [comment]
> Hi, I want to changte the "Inverter polling frequency", but which file to edit? I dont see any other option to correct the value via GUI, or did I miss something??
> 
> Thnaks To all Thomas

You switch off the "Update…" option in the integration 'System options' and add your own automation through the GUI. I have just done this using the guide here [Inverter Polling Frequency ](https://github.com/wlcrs/huawei_solar#inverter-polling-frequency)

**Comment by zapadee** [comment]
I tried to set this up for me following the tutorial, but did not manage to get it to work properly. I assumed it was because this part of the documentation was outdated. But you seemed to get it to work, so I hope you can help me out.

Given the readme example, I would expect HA to poll **only** the sensor.daily_yield values in an interval of 10 s.

`- alias: "Huawei Solar inverter data polling"
  trigger:
    - platform: time_pattern
      hours: "*"
      minutes: "*"
      seconds: "/10"
  action:
    - service: homeassistant.update_entity
      target:
        entity_id: sensor.daily_yield`

As the sensor (in my case) is called sensor.inverter_daily_yield I tried to change it accordingly but HA somehow failed to retrieve data afterwards.

For my usecase I need to update the sensor.inverter_active_power with a very low interval (preferably 1 to 2 seconds). The rest of the sensors could be updated in an interval of 30secs. 

Do I understand it right, that I would need to add every sensor to the target list in the automation with an interval of 30 secs and create another automation with an interval of e.g. 2 secs and put the sensor.inverter_active_power as a target?

**Reply by RabAusten** [reply]
Yes, you are correct. You would need to add two _Automations_ to trigger at different intervals

Based on my automations.yaml file, you'd need something like I've mocked up in the attached text file

Note: I added the basic automation through the GUI and then used _Edit in YAML_ to properly customise, so the `- id: '{Unique Automation id}'` was generated by Home Assistant. I don't know if you can create your own, but I'd recommend doing as I did, to avoid conflicts.

[ha-automation-huawei-solar-data-polling.txt](https://github.com/wlcrs/huawei_solar/files/10125785/ha-automation-huawei-solar-data-polling.txt)

**Reply by zapadee** [reply]
Works like a charm! Thanks a lot @RabAusten

**Reply by RabAusten** [reply]
Awesome! Glad I could help, @zapadee  :)

**Reply by zapadee** [reply]
Yeehaaaw. Your template really kickstarted me. I now have multiple automations up and running changing the update frequency depending on the state of the sun (sun down --> poll every minute, sun up --> i want the juice data in real time baby 👯)

**Comment by zapadee** [comment]
I have an automation up and running that polls the sensor.inverter_active_power every 2 seconds. HA only updates its value every 10 seconds though. Am I missing something?

![image](https://user-images.githubusercontent.com/72465859/205061016-1c347ba8-1168-46e2-a63b-2a711216ddc9.png)

**Reply by RabAusten** [reply]
I've seen this also, but assumed the data was collected, but the ui only updates every 8-10 secs. Not sure how you can verify this. I've only been using it for just over a week and not dug much deeper.

**Reply by zapadee** [reply]
I just took a look into my influx db where all my sensor data is stored. The UI indeed shows the realtime update of the data. In Influx a value is written every 10s. 

Which leads me to the question: Is this a hardware limitation or a limitation of the integration?

My power readings (coming from a different sensor & integration) are stored in 1s intervals as they should be.

**Comment by mariusnauj** [comment]
Hello, 

For me the same. I set up to update every 2 seconds, but on HA entity I see that it update every 10 seconds. Maybe somebody have solution?

**Comment by TMoersch** [comment]
Hi, sorry that I did nor respond in time, but I solved the (my) problem.
Under HA you need to do the follwing steps:
1. login into HA
2. open the terminal, maybe you have to install it ssh+terminal
3. change directory " cd /config/custom_components/huawei_solar"
4. open the file "const.py"
5. in line Nr. 19, change the value of UpdateInterval timedelta(seconds=10) to what ever you want
There may be other ways to do it, such like FileEditor, but this what I have done.

Think about
Huawei does not like to be too busy und hates to be checked by more than one session.
So 10 seconds worked for me perfectly, try other values, whenever you got no more results or inconsistant results, reduce the polling intervall. When I started whaching Huawei on HA I could not get often enough update, I would like to have realtime data, but this is realy not neccessary and maybe give you more questions than answers. 

In case you want more than one integration that need data vom huawei_solar (two parallel sessions, such like evcc and huawei_solar) they will both block each other. In this case you need to install modbus_proxy which is taking the parallel session and act as one common session. In this case both apllications get the correct answer and no more timeouts...
Ask me how I know ;-)

Thomas

**Comment by mariusnauj** [comment]
Hello @TMoersch ,

SO you not recommend to edit const.py UpdateInterval to 2 seconds?

**Reply by TMoersch** [reply]
I would not say that you are not recommend to edit const.py UpdateInterval to 2 seconds.....
at least you can try it, 
but I want you to know that the Huawei plattform has some limits and  I have not seen any construction that need a 2 second precision.

Do it, try it, but if you get problems go back to 10 second precision. The only thing that may happen is that you get from time to time no values. In my opinion thats worse than a 10 second precision. 

Today, now that I know that everything is working fine, I'am back to the default 30 seconds.

Thomas
